pkg/pillar/zedcloud/deferred.go

```diff
@@ -345,18 +350,29 @@ func (ctx *DeferredContext) RemoveDeferred(key string) {
 	}
 }
 
-// KickTimer kicks the timer for immediate execution
-func (ctx *DeferredContext) KickTimer() {
+// KickTimerNow kicks the timer for immediate execution
+func (ctx *DeferredContext) KickTimerNow() {
 	ctx.Ticker.TickNow()
 }
 
+// KickTimerWithinMinute kicks the timer for execution in random time
+// within a minute (reasonable time) to avoid an avalanche of messages
+// once connection being restored to the controller.
+func (ctx *DeferredContext) KickTimerWithinMinute() {
```

Comment:
As far as I understand, the interval is from 1 second to 15 minutes. So, the name `*WithinMinute()` can be confusing.

Reply:
No, noise is within a minute, that what matters. We use a second as any non zero number to have the following closed interval: [0, noise], so from 0 to minute.

Reply:
Ah, I got it, thanks

Reply:
Might make sense to add that any subsequent re-tries are exponentially backed off - future readers might have the same question as @OhmSpectator.